From 38826ee2aa08f320f39a4d3156ce049199c9839f Mon Sep 17 00:00:00 2001
From: Shuguang Gong <shuguang.gong@intel.com>
Date: Fri, 22 May 2015 13:49:59 +0800
Subject: [PATCH 404/462] ov2685: initialize 3A values when changing format

Some 3A values need to be re-initialized when format
is changed.

Change-Id: I11b40511f8ee7282b226d9aa33087c7a8ac6e067
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-27880
Signed-off-by: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/57109
Approver: Jing Tao <jing.tao@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov2685.c | 12 +++++++++---
 drivers/external_drivers/camera/drivers/media/i2c/ov2685.h |  5 +++++
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
index 81b1c41..5431b4e 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
@@ -1155,6 +1155,10 @@ static int ov2685_s_mbus_fmt(struct v4l2_subdev *sd,
 	fmt->width = ov2685_res[res_index].width;
 	fmt->height = ov2685_res[res_index].height;
 	fmt->code = V4L2_MBUS_FMT_UYVY8_1X16;
+	dev->color_effect = V4L2_COLORFX_NONE;
+	dev->wb_mode = V4L2_WHITE_BALANCE_AUTO;
+	dev->ae_lock = AE_UNLOCK;
+	dev->hot_pixel = HOT_PIXEL_OFF;
 
 	mutex_unlock(&dev->input_lock);
 	return ret;
@@ -1843,9 +1847,11 @@ static int ov2685_probe(struct i2c_client *client,
 	dev->format.code = V4L2_MBUS_FMT_UYVY8_1X16;
 	dev->sd.entity.ops = &ov2685_entity_ops;
 	dev->sd.entity.type = MEDIA_ENT_T_V4L2_SUBDEV_SENSOR;
-	dev->ae_lock = 0;
-	dev->color_effect = 0;
-	dev->hot_pixel = 0;
+	dev->color_effect = V4L2_COLORFX_NONE;
+	dev->wb_mode = V4L2_WHITE_BALANCE_AUTO;
+	dev->ae_lock = AE_UNLOCK;
+	dev->hot_pixel = HOT_PIXEL_OFF;
+
 	ret = media_entity_init(&dev->sd.entity, 1, &dev->pad, 0);
 	if (ret)
 		ov2685_remove(client);
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
index 198df8e..9bbf56c 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
@@ -120,6 +120,11 @@
 #define MIN_SHUTTER		0
 #define MIN_GAIN		0
 
+#define AE_UNLOCK	0
+#define AE_LOCK		1
+#define HOT_PIXEL_OFF	0
+#define HOT_PIXEL_ON	1
+
 /* OV2685_DEVICE_ID */
 #define OV2685_MOD_ID		0x2685
 
-- 
1.9.1

